### ISSUE TITLE
Add support for -t / --team on teams

### ISSUE THREAD

**Issue by raulb** [heroku/heroku-apps#140]
There are other commands like `heroku members` that do support `-t` and `--team` and even encourages their use. 

I think switching between commands that support this flag and others that doesn't, it might be really confusing to the user:

e.g.:

```
› heroku members -o my-team
member@email.com    member
admin@email.com     admin
 ▸    my-team is a Heroku Team
 ▸    Heroku CLI now supports Heroku Teams.
 ▸    Use -t or --team for teams like my-team

› heroku members -t my-team
adam@grailed.com    member
admin@email.com     admin

› heroku apps -t my-team
 ▸    Error: Unexpected flag -t
 ▸    Usage: heroku apps
 ▸
 ▸    This flag is invalid for this command. Here are the accepted flags:
 ▸     -A, --all           # include apps in all organizations
 ▸     -o, --org ORG       # organization to use
 ▸     -p, --personal      # list apps in personal account when a default org is set
 ▸     -s, --space SPACE   # filter by space
 ▸     --json              # output in json format
 ▸
 ▸    See more information with heroku apps --help
```

cc @simpsoka @ike-delorenzo

**Comment by dickeyxxx** [comment]
It would be ideal to get a list of all the commands we need to add it to

**Comment by raulb** [comment]
@dickeyxxx yup, I agree. Now that I'm in DevEx, I influenced the inclusion of this flag to existing commands (see https://github.com/heroku/heroku-pipelines/pull/43/commits/65d3e0844e2af0985ff7775c1c01f4fd04c26ba7), but we should probably collect, from core plugins what commands currently operate with `-o`. Anything that meets that criteria, should probably also support `t` / `--team`. 

Update, just took a glance at the [current CLI core plugins](https://github.com/heroku/cli/blob/5218800d78a7d9457afd55d3bc298982e82deb66/package.json#L6-L18), and AFAICT, only `heroku-apps` and `heroku-spaces` operate with `--org`. For `heroku-spaces`, adding `-t` support is not necessary since this is an Enterprise feature only (no teams), so it leaves us with just `heroku-apps`. 

My guess is that `heroku apps` really is the most common one, but if I think of new ones, I'll make sure I'll open issues where they belong.
